query-engine/query-engine-wasm/src/wasm/logger.rs

```diff
@@ -43,17 +43,26 @@ impl Logger {
             FilterExt::boxed(log_level)
         };
 
-        let log_callback = CallbackLayer::new(log_callback);
-        let layer = log_callback.with_filter(filters);
+        let log_layer = CallbackLayer::new(log_callback).with_filter(filters);
+
+        let exporter = Exporter::new();
+
+        let tracing_layer = enable_tracing
+            .then(|| telemetry::layer(exporter.clone()).with_filter(telemetry::filter::user_facing_spans()));
 
         Self {
-            dispatcher: Dispatch::new(Registry::default().with(layer)),
+            dispatcher: Dispatch::new(Registry::default().with(tracing_layer).with(log_layer)),
+            exporter,
         }
     }
 
     pub fn dispatcher(&self) -> Dispatch {
         self.dispatcher.clone()
     }
+
+    pub fn exporter(&self) -> Exporter {
+        self.exporter.clone()
+    }
```

Comment:
These two methods are never called. Can we remove them?

Reply:
what do you mean? they are called in most methods in the engine

Reply:
Aargh, I was misleaded by a broken Rust Analyser run. Please ignore this comment (which I repeated in other `logger.rs` files).

<img width="1420" alt="Screenshot 2024-12-16 at 11 43 22" src="https://github.com/user-attachments/assets/61b6dc71-dfa9-45be-b876-a959781fa9d5" />